### PR TITLE
[python] Use C++ `DataFrame` bindings for create and writes

### PR DESCRIPTION
**Pull request by nguyenv**
**Issue and/or context:** #2427

Part of larger branch https://github.com/single-cell-data/TileDB-SOMA/pull/2367

**Changes:**

Remove TileDB-Py usage for `DataFrame` by using `create` and `write` C++ bindings.

### CODE REVIEW

File: apis/python/src/tiledbsoma/_tdb_handles.py [hunk 1]
```diff
@@ -155,7 +156,9 @@ def open(
                     handle._do_initial_reads(auxiliary_reader)
             else:
                 handle._do_initial_reads(tdb)
-        except tiledb.TileDBError as tdbe:
+
+        # TODO macOS throws RuntimeError
```
Comment (johnkerl) [inline apis/python/src/tiledbsoma/_tdb_handles.py:160]: ✨

**Review by johnkerl — APPROVED** [review]

**Comment by eddelbuettel** [comment]
This once again whitespace changed nanoarrow.h which, as an external file, is not really needed -- and makes comparison to upstream more difficult.

**Review by jp-dark — APPROVED** [review]
Looks good to me.

**Review by ryan-williams — APPROVED** [review]
lgtm, thanks for the walk-through yesterday!

**Pull request merged**